src/main/scala/tilelink/Parameters.scala

```diff
@@ -534,48 +537,81 @@ class TLSlavePortParameters private(
   // Does this Port manage this ID/address?
   def containsSafe(address: UInt) = findSafe(address).reduce(_ || _)
 
-  private def supportHelper(
+  private def addressHelper(
       safe:    Boolean,
-      member:  TLManagerParameters => TransferSizes,
+      member:  TLSlaveParameters => TransferSizes,
       address: UInt,
       lgSize:  UInt,
-      range:   Option[TransferSizes]): Bool = {
+      range:   Option[TransferSizes]): Bool = { // What does the range argument do?
+    // trim is just processing the range argument
     def trim(x: TransferSizes) = range.map(_.intersect(x)).getOrElse(x)
     // groupBy returns an unordered map, convert back to Seq and sort the result for determinism
-    val supportCases = groupByIntoSeq(managers)(m => trim(member(m))).map { case (k, vs) =>
+    // groupByIntoSeq is turning slaves into trimmed membership sizes
+    // We are grouping all the slaves by their transfer size where
+    // if they support the trimmed size then
+    // member is the type of transfer that you are looking for (What you are trying to filter on)
+    // When you consider membership, you are trimming the sizes to only the ones that you care about
+    // you are filtering the slaves based on both whether they support a particular opcode and the size
+    // Grouping the slaves based on the actual transfer size range they support
+    // intersecting the range and checking their membership
+    // FOR SUPPORTCASES instead of returning the list of slaves,
+    // you are returning a map from transfer size to the set of
+    // address sets that are supported for that transfer size
+
+    // find all the slaves that support a certain type of operation and then group their addresses by the supported size
+    // for every size there could be multiple address ranges
+    // safety is a trade off between checking between all possible addresses vs only the addresses
+    // that are known to have supported sizes
+    // the trade off is 'checking all addresses is a more expensive circuit but will always give you
+    // the right answer even if you give it an illegal address'
+    // the not safe version is a cheaper circuit but if you give it an illegal address then it might produce the wrong answer
+    // fast presumes address legality
+    val supportCases = groupByIntoSeq(slaves)(m => trim(member(m))).map { case (k: TransferSizes, vs: Seq[TLSlaveParameters]) =>
       k -> vs.flatMap(_.address)
     }
+    // the rest of this function is just processing that more
+    // safe produces a circuit that compares against all possible addresses
+    // whereas fast presumes that the address is legal
+    // It creates an address decoder that only works if the address is supported
     val mask = if (safe) ~BigInt(0) else AddressDecoder(supportCases.map(_._2))
+    // simplified...?
+    // simplified is doing some unification with the addresses to combine them with the mask
```

Comment:
// Simplified crates the most concise possible representation of each cases' address sets based on the mask.